plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1470,8 +1470,14 @@ private void finishOptimize(ConnectorSession session, DeltaLakeTableExecuteHandl
 
     private boolean allowWrite(ConnectorSession session, DeltaLakeTableHandle tableHandle)
     {
-        boolean requiresOptIn = transactionLogWriterFactory.newWriter(session, tableHandle.getLocation()).isUnsafe();
-        return !requiresOptIn || unsafeWritesEnabled;
+        try {
+            boolean requiresOptIn = transactionLogWriterFactory.newWriter(session, tableHandle.getLocation()).isUnsafe();
```

Comment:
IAE is a rather generic exception.

Catching the exception may hide other problems.

From the commit content is not quite clear what kind of problem is being addressed - there is no associated test for this change.